test/1.4/legacy/lenient_typechecking.dml

```diff
@@ -0,0 +1,42 @@
+/*
+  © 2024 Intel Corporation
+  SPDX-License-Identifier: MPL-2.0
+*/
+
+dml 1.4;
+
+template t {
+    shared method sm(const int *p) -> (const int *) default { return p; }
+    shared method m(const int *p) -> (const int *);
+    method n(const int *p) -> (const int *) default { return p; }
+}
+
+template u is t {
+    /// ERROR EMETH
+    shared method sm(int *p) -> (int *) { return p; }
+}
+
+group g is u {
+    /// ERROR EMETH
+    method m(int *p) -> (int *) { return p; }
+
+    /// ERROR EMETH
+    method n(int *p) -> (int *) { return p; }
+}
+
+header %{
+    #define MACRO(a,b,c) ((void)0)
+%}
+
```

Comment:
We probably want more test coverage of `cmp`, e.g. whether `uint16` is compatible with `bitfields 16`. Perhaps something for `types_test.py`?

Reply:
Might also be nice with a ctree_test-style test to get reasonable coverage of the invariant that DML type compatibility implies C type compatibility.

Reply:
_done_